packages/lyra/src/lyra.ts

```diff
@@ -4,12 +4,28 @@ import * as fastq from "fastq";
 import { Trie } from "./prefix-tree/trie";
 import * as ERRORS from "./errors";
 import { tokenize } from "./tokenizer";
-import { formatNanoseconds, getNanosecondsTime } from "./utils";
+import {
+  binarySearch,
+  formatNanoseconds,
+  getNanosecondsTime,
+  setIntersection,
+  setSubtraction,
+} from "./utils";
 import { Language, SUPPORTED_LANGUAGES } from "./stemmer";
-import type { ResolveSchema, SearchProperties } from "./types";
+import type {
+  Nullable,
+  NumberComparison,
+  ResolveSchema,
+  SearchProperties,
+  WhereParams,
+} from "./types";
+import { SortedQueue } from "./sorted-queue";
+import { QueueNode } from "./sorted-queue/node";
 
 export type PropertyType = "string" | "number" | "boolean";
 
+export const allowedNumericComparison = [">=", ">", "=", "<", "<="] as const;
```

Comment:
Why not `!=`?

Reply:
We started with the simplest one, later we can implement things like INCLUDES, LIKE, OR and !=.

I excluded that because I think that is harder to implement now in this PR since I need to do a linear search. We can do that in the future.